src/Felix/Instances/Setoid/Raw.agda

```diff
@@ -0,0 +1,84 @@
+{-# OPTIONS --safe --without-K  #-}
+
+module Felix.Instances.Setoid.Raw where
+
+open import Data.Product using (_,_; _,′_; curry′; uncurry′; ∃₂; proj₁; proj₂)
+open import Data.Product.Function.NonDependent.Setoid using (<_,_>ₛ; proj₁ₛ; proj₂ₛ)
+open import Data.Sum using ([_,_]; inj₁; inj₂)
+open import Data.Sum.Function.Setoid using ([_,_]ₛ; inj₁ₛ; inj₂ₛ)
+open import Data.Unit.Polymorphic using (tt)
+open import Function using (Func) renaming (_∘_ to _∘ᵈ_)
+open import Function.Construct.Composition as Comp
+open import Function.Construct.Constant as Const
+open import Function.Construct.Identity as Id
+open import Level using (_⊔_; suc)
+open import Relation.Binary using (Setoid)
+
+open import Function.Construct.Setoid using (setoid)
+open import Felix.Equiv using (Equivalent)
+open import Felix.Raw
+open import Felix.Instances.Setoid.Type
+  using (module setoid-instances; _⟶_; cong; _⟨$⟩_) public
+open setoid-instances public
+
+module setoid-raw-instances where instance
+
+  equivalent : ∀ {c ℓ} → Equivalent (c ⊔ ℓ) {obj = Setoid c ℓ} _⟶_
+  equivalent = record
+    { _≈_ = λ {From} {To} → Setoid._≈_ (setoid From To)
+    ; equiv = λ {From} {To} → Setoid.isEquivalence (setoid From To)
+    }
```

Comment:
Thanks. I was hoping for a general mapping from `Setoid` to `Equivalent`, but now I realize that such a mapping would be tricky, since the setoid carrier might not have the required shape.

Reply:
I did try that for a bit, but couldn't figure it out